packages/overlay/src/topLayerOverTransforms.ts

```diff
@@ -79,9 +79,11 @@ export const topLayerOverTransforms = (): Middleware => ({
             ) {
                 const css = getComputedStyle(containingBlock);
                 overTransforms =
-                    css.transform !== 'none' || css.filter
-                        ? css.filter !== 'none'
-                        : false;
+                    css.transform !== 'none' ||
+                    (css.filter ? css.filter !== 'none' : false) ||
+                    ['paint', 'layout', 'strict', 'content'].some((value) =>
+                        (css.contain || '').includes(value)
+                    );
```

Comment:
This feels slightly esoteric to me and I'm worried we may forget what this is for 😅 It seems efficient, but maybe could benefit from a clarifying comment? Curious what others think.